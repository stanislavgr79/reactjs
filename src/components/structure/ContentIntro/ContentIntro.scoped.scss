.content-intro {
    min-height: calc(100vh - 70px);
    background-color: #232323;
}