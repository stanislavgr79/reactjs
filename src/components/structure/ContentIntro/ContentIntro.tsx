import React, { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import { AppState } from '../../../redux/redux-store';

import NavBar from '../../container/Navbar';
import MoviesList from '../../container/MoviesList';
import ErrorBoundary from '../../../components/plugins/ErrorBoundary';
import withLoading from '../../../components/plugins/WithLoading';

import './ContentIntro.scoped.scss';

const MovieListWithLoading = withLoading(MoviesList);

export default function ContentIntro(): JSX.Element {
  const store = useSelector((store: AppState) => {
    return {
      moviesIsLoading: store.moviesStore.moviesIsLoading,
    };
  });
  const [isLoading, setIsloading] = useState(store.moviesIsLoading);

  useEffect(() => {
    const id = setTimeout(() => {
      if (isLoading) {
        return;
      }
      setIsloading(false);
    }, 1500);
    return () => clearTimeout(id);
  }, [isLoading]);

  return (
    <div className="content content-intro">
      <NavBar />
      <ErrorBoundary>
        <MovieListWithLoading isLoading={!isLoading} />
      </ErrorBoundary>
    </div>
  );
}
