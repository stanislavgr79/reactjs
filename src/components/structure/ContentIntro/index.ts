import ContentIntro from './ContentIntro';

export default ContentIntro;
