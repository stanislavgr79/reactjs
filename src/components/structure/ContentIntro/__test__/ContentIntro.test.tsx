import React from 'react';
import { render } from '@testing-library/react';

import ContentIntro from '../ContentIntro';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

jest.mock('../../../../components/container/NavBar', () => {
  return {
    __esModule: true,
    default: () => {
      return <div className="nav-bar container-md"></div>;
    },
  };
});
jest.mock('../../../../components/container/MoviesList', () => {
  return {
    __esModule: true,
    default: () => {
      return <div className="movies-list"></div>;
    },
  };
});

interface PropsErrorBoundary {
  children: React.ReactFragment;
}
jest.mock('../../../plugins/ErrorBoundary', () => ({ children }: PropsErrorBoundary) => children);

describe('ContentIntro', () => {
  test('snapshot renders: movies list', () => {
    const initialState = { moviesStore: { newPage: true, dataStatus: 'idle' } };
    const mockStore = configureStore();
    const store = mockStore(initialState);

    const tree = render(
      <Provider store={store}>
        <ContentIntro />
      </Provider>,
    );

    expect(tree.baseElement).toMatchSnapshot();
    expect(tree.baseElement).toContainHTML('<div class="movies-list">');
  });

  test('snapshot renders: movies are loading', () => {
    const initialState = { moviesStore: { newPage: false, dataStatus: 'idle' } };
    const mockStore = configureStore();
    const store = mockStore(initialState);

    const tree = render(
      <Provider store={store}>
        <ContentIntro />
      </Provider>,
    );

    expect(tree.baseElement).toMatchSnapshot();
    expect(tree.baseElement).toContainHTML('<div class="content content-intro">');
    expect(tree.getByText('Just a moment... Movies are loading...')).toBeInTheDocument();
  });

  test('snapshot renders: movies list', () => {
    const initialState = { moviesStore: { newPage: false, dataStatus: 'success' } };
    const mockStore = configureStore();
    const store = mockStore(initialState);

    const tree = render(
      <Provider store={store}>
        <ContentIntro />
      </Provider>,
    );

    expect(tree.baseElement).toMatchSnapshot();
    expect(tree.baseElement).toContainHTML('<div class="movies-list">');
  });
});
