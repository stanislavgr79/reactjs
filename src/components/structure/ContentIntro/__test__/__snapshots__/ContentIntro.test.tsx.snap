// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentIntro snapshot renders: movies are loading 1`] = `
<body>
  <div>
    <div
      class="content content-intro"
    >
      <div
        class="nav-bar container-md"
      />
      <h2>
        Just a moment... Movies are loading...
      </h2>
    </div>
  </div>
</body>
`;

exports[`ContentIntro snapshot renders: movies list 1`] = `
<body>
  <div>
    <div
      class="content content-intro"
    >
      <div
        class="nav-bar container-md"
      />
      <div
        class="movies-list"
      />
    </div>
  </div>
</body>
`;

exports[`ContentIntro snapshot renders: movies list 2`] = `
<body>
  <div>
    <div
      class="content content-intro"
    >
      <div
        class="nav-bar container-md"
      />
      <div
        class="movies-list"
      />
    </div>
  </div>
</body>
`;
